From 5778986ebc343f52bc337371d9ba82a51bee77a0 Mon Sep 17 00:00:00 2001
From: Suneel Garapati <sgarapati@marvell.com>
Date: Mon, 11 Nov 2019 14:18:43 -0800
Subject: [PATCH 0437/1921] spi: octeontx2: set tritx in config register

With 9xx chips, bus width is expanded from x1 to x4, in order
to avoid driving of IO1/2/3 during bus idle, tritx must be
set to 1 to tristate these signals otherwise with some flash
parts, like Micron/Cypress, IO3 has multiplexed functionality
of HOLD and treats driving low as valid causing incorrect bus
operations to the flash parts.

Change-Id: Ia6c1f9eaa7cd86a63eadaf5f715f1f459863b384
Signed-off-by: Suneel Garapati <sgarapati@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/18707
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/26932
Reviewed-by: Suneel Garapati <sgarapati@caviumnetworks.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/spi/spi-octeontx2.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/spi/spi-octeontx2.c b/drivers/spi/spi-octeontx2.c
index 46f598833691..e797b00e9e61 100644
--- a/drivers/spi/spi-octeontx2.c
+++ b/drivers/spi/spi-octeontx2.c
@@ -81,6 +81,7 @@ static int octeontx2_spi_do_transfer(struct octeontx2_spi *p,
 	mpi_cfg.s.wireor = (mode & SPI_3WIRE) ? 1 : 0;
 	mpi_cfg.s.idlelo = cpha != cpol;
 	mpi_cfg.s.cslate = cpha ? 1 : 0;
+	mpi_cfg.s.tritx = 1;
 	mpi_cfg.s.enable = 1;
 	mpi_cfg.s.cs_sticky = 1;
 	mpi_cfg.s.legacy_dis = 1;
-- 
2.31.1

